src/thumbnailer_shared.lua

```diff
@@ -236,6 +236,10 @@ function Thumbnailer:get_thumbnail_size()
     if not (video_width and video_height) then
         return nil
     end
+    local rotate = mp.get_property_number("video-params/rotate")
+    if rotate ~= nil and rotate % 180 == 90 then
```

Comment:
```suggestion
    local rotate = mp.get_property_number("video-params/rotate", 0)
    if rotate % 180 == 90 then
```
It's cleaner to have a default and avoid checking for `nil`.

Reply:
> This is indeed what happens with my test file if the thumbnails are generated before the video is rotated

[Some insight from thumbfast](https://github.com/po5/thumbfast/blob/4241c7daa444d3859b51b65a39d30e922adb87e9/thumbfast.lua#L393-L395), if we're missing rotate info on the first run, we queue another check in 0.05s (in effect the next even loop).  
This is obviously hacky, it's a leftover of when thumbfast queried properties when needed instead of watching for changes and caching the value.  
I need to rewrite this to call `info()` within `watch_changes()` when `video-params/rotate` changes, that way it's no longer racy. Though this logic may not be needed at all anymore as [we delay announcing until `video-out-params` is available](https://github.com/po5/thumbfast/blob/4241c7daa444d3859b51b65a39d30e922adb87e9/thumbfast.lua#L762).

We should do a similar thing in mpv_thumbnail_script, return `nil` in `get_thumbnail_size()` until `video-params/rotate` is available (is it guaranteed to eventually exist?). There is already a check for `video-dec-params/dw`, `video-dec-params/dh`. The issue may go away if we make the check more robust.